backend/routes/events.js

```diff
@@ -1,40 +1,41 @@
 var express = require('express');
 var router = express.Router();
+const passport = require('passport');
 
 //Import Controllers
 const EventController = require('../controllers/EventController');
 
 router.get('/', EventController.getAllEvents);
 router.get('/getEventbyCreator/:id', EventController.getEventbyCreator);
 
-router.post('/', EventController.addEvent);
+router.post('/', passport.authenticate('jwt', {session: false}), EventController.addEvent);
 
 //Media endpoints
 router.post('/:id/media', EventController.addMedia);
 router.get('/:id/media', EventController.getMedia);
-router.put('/:id/media/:mediaId', EventController.updateMedia);
-router.delete('/:id/media/:mediaId', EventController.deleteMedia);
+router.put('/:id/media/:mediaId', passport.authenticate('jwt', {session: false}), EventController.updateMedia);
+router.delete('/:id/media/:mediaId', passport.authenticate('jwt', {session: false}), EventController.deleteMedia);
 
 
 // Adds an attendance to an event
-router.post('/:id/attendance', EventController.addAttendance);
-router.get('/:id/attendance', EventController.getAttendance);
-router.put('/:id/attendance', EventController.updateAttendance);
+router.post('/:id/attendance', passport.authenticate('jwt', {session: false}), EventController.addAttendance);
+router.get('/:id/attendance', passport.authenticate('jwt', {session: false}), EventController.getAttendance);
```

Comment:
Is this **get** still needs to be private to users?

Reply:
We shouldn't share information about where users will be to non members. I did this intuitively.

Reply:
We shouldn't share information about where users will be to non members. I did this intuitively.

Reply:
Is it still possible for us to retrieve **attendance counts**?